### PR TITLE
[fix] FMC RtAlias boot flow raising errors when signing the certificate.

### PR DESCRIPTION
**Pull request by rusty1968**
This commit fixes the RtAlias derivation flow and cleans up unused code.  It also adds unit tests for caliptra_common::hand_off module.

### CODE REVIEW

**Review by jlmahowa-amd — APPROVED** [review]

File: fmc/README.md [hunk 1]
```diff
@@ -239,6 +239,23 @@ The following list of steps are to be performed by FMC on each boot when ROM jum
 
 <center>
 
+**Pre-Conditions:**
```
Comment (FerralCoder) [inline fmc/README.md:242]: Can you also add the same table showing the state AFTER FMC runs?

File: common/src/hand_off.rs [hunk 2]
```diff
@@ -23,6 +23,9 @@ impl HandOffDataHandle {
        vault, set_vault : 15, 12;
        reserved, _: 31, 16;
     }
+    pub fn is_valid(&self) -> bool {
+        self.0 != u32::MAX
```
[the remark this answers is not in the document]
Comment (rusty1968) [reply common/src/hand_off.rs:27]: Invalid address is for an address, not for a handle.

File: fmc/README.md (hunk 1 above)
Comment (rusty1968) [reply fmc/README.md:242]: Done

File: common/src/hand_off.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (FerralCoder) [reply common/src/hand_off.rs:27]: sorry, copied wrong one, meant FHT_INVALID_HANDLE. Avoid magic numbers and use defines wherever you can

[the remark this answers is not in the document]
Comment (rusty1968) [reply common/src/hand_off.rs:27]: fht invalid handle is for the whole type.

File: fmc/src/flow/rt_alias.rs [hunk 3]
```diff
@@ -226,23 +230,32 @@ impl RtAliasLayer {
...
+        // AliasRtTbsDigest = sha384_digest(AliasRtTbs) AliaRtTbsCertSig = ecc384_sign(KvSlot5, AliasFmcTbsDigest)
+
         let sig = Crypto::ecdsa384_sign(env, auth_priv_key, tbs.tbs());
         let sig = okref(&sig)?;
+        // Clear the authority private key
+        cprintln!(
+            "[alias rt] Erasing AUTHORITY.KEYID = {}",
+            auth_priv_key as u8
+        );
+        env.key_vault.erase_key(auth_priv_key)?;
```
Comment (jhand2) [inline fmc/src/flow/rt_alias.rs:248]: Should it be erased or locked? On impactless update reset I would expect FMC to need to use this key again and I'm not sure if ROM will re-derive it.

File: fmc/src/hand_off.rs [hunk 4]
```diff
@@ -230,8 +297,9 @@ impl HandOff {
         self.fht.rt_cdi_kv_hdl = Self::rt_cdi_store(&out);
         self.fht.rt_priv_key_kv_hdl = Self::rt_priv_key_store(&out);
         self.fht.rt_dice_pub_key = out.subj_key_pair.pub_key;
-        self.fht.fmc_cdi_kv_hdl = FHT_INVALID_HANDLE;
-        self.fht.fmc_priv_key_kv_hdl = FHT_INVALID_HANDLE;
         Ok(())
     }
+    pub fn is_valid(&self) -> bool {
```
Comment (jhand2) [inline fmc/src/hand_off.rs:302]: Could you add a comment explaining in what way this is checking validity of the FHT? IIUC, it's just making sure the RT CDI and private key are present in KeyVault. Is that correct?

Comment (rusty1968) [reply fmc/src/hand_off.rs:302]: Correct. Upon exit, before FMC hands execution off to RT, it needs to make sure the dice output has been produced.

File: fmc/src/flow/rt_alias.rs (hunk 3 above)
Comment (rusty1968) [reply fmc/src/flow/rt_alias.rs:248]: I was following what ROM is doing in the FMC Alias flow. : https://github.com/chipsalliance/caliptra-sw/blob/main/rom/dev/src/flow/cold_reset/fmc_alias.rs#L176

I am going to lock instead.

**Review by jhand2 — APPROVED** [review]

File: fmc/src/hand_off.rs [hunk 5]
```diff
@@ -216,22 +283,25 @@ impl HandOff {
...
-        HandOffDataHandle(((Vault::KeyVault as u32) << 12) | output.cdi as u32)
+    fn rt_cdi_store(rt_cdi: KeyId) -> HandOffDataHandle {
+        HandOffDataHandle(((Vault::KeyVault as u32) << 12) | rt_cdi as u32)
     }
 
-    fn rt_priv_key_store(output: &DiceOutput) -> HandOffDataHandle {
-        HandOffDataHandle(((Vault::KeyVault as u32) << 12) | output.subj_key_pair.priv_key as u32)
+    fn rt_priv_key_store(rt_priv_key: KeyId) -> HandOffDataHandle {
+        HandOffDataHandle(((Vault::KeyVault as u32) << 12) | rt_priv_key as u32)
     }
```
Comment (jhand2) [inline fmc/src/hand_off.rs:292]: nit: This can be a generic function `get_keyid_handle` right? It looks like they're doing the same thing

**Review by jhand2 — APPROVED** [review]

File: fmc/src/hand_off.rs (hunk 5 above)
Comment (rusty1968) [reply fmc/src/hand_off.rs:292]: Created an issue as follow up

**Pull request merged**